src/python/WMCore/MicroService/Unified/Transferor.py

```diff
@@ -170,229 +111,169 @@ def transferor(self, req_status='assigned'):
-            # uncomment when we'll deal with real transfers
-            #self.checkStatus(request)
+            requestRecords = requestsInfo(requestRecords, self.svc, self.logger)
+        except Exception as err:  # general error
+            msg = '### transferor error: %s' % str(err)
+            self.logger.error(msg)
+            requestRecords = []
+        # process all requests
+        for req in requestRecords:
+            reqName = req['name']
+            # perform transfer
+            tid = self.transferRequest(req)
+            if tid:
+                # Once all transfer requests were successfully made, update: assigned -> staging
+                self.logger.debug("### transfer request for %s successfull", reqName)
+                self.change(req, 'staging', '### transferor')
+            # if there is nothing to be transferred (no input at all),
+            # then update the request status once again staging -> staged
+            # self.change(req, 'staged', '### transferor')
 
     def stop(self):
         "Stop MSManager"
         # stop MSTransferorMonit thread
         self.ms_monit.stop()
         # stop MSTransferor thread
-        self.thr.stop() # stop checkStatus thread
+        self.thr.stop()  # stop checkStatus thread
         status = self.thr.running()
         return status
 
-    def transferRequest(self, request, rdict):
+    def transferRequest(self, req):
         "Send request to Phedex and return status of request subscription"
-	datasets = rdict.get('datasets', [])
-	sites = rdict.get('sites', [])
+        datasets = req.get('datasets', [])
+        sites = req.get('sites', [])
         if datasets and sites:
-	    subscription = PhEDExSubscription(datasets, sites, self.group)
-	    self.logger.debug("### add subscription %s for request %s" % (subscription, request))
+            subscription = PhEDExSubscription(datasets, sites, self.group)
+            self.logger.debug("### add subscription %s for request %s", subscription, req)
             # TODO: implement how to get transfer id
             tid = hashlib.md5(str(subscription)).hexdigest()
-	    # TODO: when ready enable submit subscription step
-	    # self.phedex.subscribe(subscription)
+            # TODO: when ready enable submit subscription step
+            # self.phedex.subscribe(subscription)
             return tid
 
     def getTransferIDsDoc(self):
         """
         Get transfer ids document from backend. The document has the following form:
         https://gist.github.com/amaltaro/72599f995b37a6e33566f3c749143154
-	{"wf_A": {"timestamp": 0000
-		  "primary": {"dset_1": ["list of transfer ids"]},
-		  "secondary": {"PU_dset_1": ["list of transfer ids"]},
-	 "wf_B": {"timestamp": 0000
-		  "primary": {"dset_1": ["list of transfer ids"],
-			      "parent_dset_1": ["list of transfer ids"]},
-		  "secondary": {"PU_dset_1": ["list of transfer ids"],
-				"PU_dset_2": ["list of transfer ids"]},
+        {"wf_A": {"timestamp": 0000
```

Comment:
this data structure is no longer going to be used. We decided to adopt option C from the gist.